repository_service_tuf/cli/artifact/download.py

```diff
@@ -0,0 +1,274 @@
+
+from click import Context
+from tuf.api.exceptions import DownloadError, RepositoryError
+from tuf.ngclient import Updater, UpdaterConfig
+
+from repository_service_tuf.cli import click, console
+from repository_service_tuf.cli.artifact import artifact
+
+
+def decode_trusted_root(root) -> str:
+    root_encoded = bytes(root, "utf-8")
+    return base64.b64decode(root_encoded).decode("utf-8")
+
+
+def build_metadata_dir(metadata_url: str) -> str:
+    """build a unique and reproducible directory name for the repository url"""
+    name = sha256(metadata_url.encode()).hexdigest()[:8]
+    return f"{Path.home()}/.local/share/rstuf/{name}"
+
+
+def init_tofu(metadata_url: str, root_url: Optional[str]) -> bool:
+    """Initialize local trusted metadata (Trust-On-First-Use) and create a
+    directory for downloads"""
+    metadata_dir = build_metadata_dir(metadata_url)
+
+    os.makedirs(metadata_dir, exist_ok=True)
+
+    if not root_url:
+        root_url = f"{metadata_url}/1.root.json"
+    try:
+        parsed_root_url = urlparse(root_url)
+        if parsed_root_url.scheme and parsed_root_url.netloc:
+            request.urlretrieve(root_url, f"{metadata_dir}/root.json")  # nosec
+        else:
+            console.print(  # pragma: no cover
+                f"Failed to parse {root_url}: ",
+                f"{parsed_root_url.scheme}, {parsed_root_url.netloc}",
+            )
+    except OSError:  # pragma: no cover
+        console.print(f"Failed to download initial root from {root_url}")
+        return False  # pragma: no cover
+
+    console.print(
+        f"Trust-on-First-Use: Initialized new root in {metadata_dir}"
+    )
+    return True
+
+
+def perform_tuf_ngclient_download_artifact(
+    metadata_url: str,
+    metadata_dir: str,
+    artifacts_url: str,
+    artifact_name: str,
+    download_dir: str,
+    config: UpdaterConfig,
+) -> bool:
+    try:
+        updater = Updater(
+            metadata_dir=metadata_dir,
+            metadata_base_url=metadata_url,
+            target_base_url=artifacts_url,
+            target_dir=download_dir,
+            config=config,
+        )
+        updater.refresh()
+
+        # Annotating, as it's an external method we don't want to test
+        info = updater.get_targetinfo(artifact_name)  # pragma: no cover
+
+        if info is None:  # pragma: no cover
+            console.print(f"Artifact {artifact_name} not found")
+            return False
+
+        path = updater.find_cached_target(info)  # pragma: no cover
+        if path:  # pragma: no cover
+            console.print(f"Artifact is available in {path}")
+            return True
+
+        path = updater.download_target(info)  # pragma: no cover
+        console.print(  # pragma: no cover
+            f"Artifact downloaded and available in {path}"
+        )
+
+    except (OSError, RepositoryError, DownloadError) as e:
+        console.print(f"Failed to download artifact {artifact_name}: {e}")
+        return False
```

Comment:
This should be an exception instead.
We don't need to return anything from this function. It doesn't make any boolean check and if it didin't raise anything it succeded.